src/shapes/polyline.class.ts

```diff
@@ -79,36 +80,41 @@ import { makeBoundingBoxFromPoints } from '../util/misc/boundingBoxFromPoints';
         return projectStrokeOnPoints(this.points, this, true);
       },
 
-      _setPositionDimensions: function (options) {
-        options || (options = {});
-        var calcDim = this._calcDimensions(options),
-          correctLeftTop,
-          correctSize = this.exactBoundingBox ? this.strokeWidth : 0;
-        this.width = calcDim.width - correctSize;
-        this.height = calcDim.height - correctSize;
-        if (!options.fromSVG) {
-          correctLeftTop = this.translateToGivenOrigin(
-            {
-              // this looks bad, but is one way to keep it optional for now.
-              x: calcDim.left - this.strokeWidth / 2 + correctSize / 2,
-              y: calcDim.top - this.strokeWidth / 2 + correctSize / 2,
-            },
+      _setPositionDimensions: function ({ left, top, fromSVG } = {}) {
+        const bbox = this._calcDimensions(),
+          strokeCorrection = new Point()
+            .scalarAdd(this.strokeWidth)
+            .divide(
+              this.strokeUniform
+                ? new Point(this.scaleX, this.scaleY)
+                : new Point(1, 1)
+            );
+        this.width = bbox.width - strokeCorrection.x;
```

Comment:
Here the 0 size = -1 edge case I thought of wrapping with Math.max
I wonder if it's related somehow to the 1px diff in tests

Reply:
solved!